### PR TITLE
[WIP][SPARK-35917][SHUFFLE][CORE]Disable push-based shuffle feature to prevent it from being used

### PR DESCRIPTION
**Pull request by otterc**
This is WIP because https://github.com/apache/spark/pull/33118#issuecomment-870084009

### What changes were proposed in this pull request?
Push-based shuffle is partially merged in apache master but some of the tasks are still incomplete. Since 3.2 is going to cut soon, we will not be able to get the pending tasks reviewed and merged. Few of the pending tasks make protocol changes to the push-based shuffle protocols, so we would like to prevent users from enabling push-based shuffle both on the client and the server until push-based shuffle implementation is complete. 
We can prevent push-based shuffle to be used by throwing UnsupportedOperationException both on the client and the server when the user tries to enable it.

### Why are the changes needed?
The change is needed to prevent users from trying out push-based shuffle until it is complete.

### Does this PR introduce _any_ user-facing change?
Yes. It will prevent users to try out push-based shuffle until it is complete.

### How was this patch tested?
It is a straightforward change that prevents users from enable push-based shuffle so haven't added a UT for it.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala
Comment (dongjoon-hyun) [inline core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala:23]: This will break scala style.

Comment (otterc) [reply core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala]: I just fixed it.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
It's possible, but this looks like an overkill a little to me. Why don't we update the config doc instead? I believe that's more proper and correct location to do this.

cc @mridulm

**Comment by mridulm** [comment]
@dongjoon-hyun There are two ongoing correctness PR's which require protocol changes (for push based shuffle) in order to fix the issues. I am expecting them to be reviewed and merged before 3.2.
But in case they do not make it into 3.2, we would need to make sure that in future a 3.3 ESS/client does not badly interact with a 3.2 ESS/client.

This PR would be merged only if those other two pr's (#33078 and #33034) do not make it into 3.2
Thoughts ?

Do we want to make this WIP to convey this intent Dongjoon ?

**Comment by otterc** [comment]
@dongjoon-hyun @mridulm I have updated the docs of the configs on both the client and sever as well with 
`Push-based shuffle is not yet supported.`
I will change this to WIP.

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you for the detailed information. Are we going to merge this to the `master` and revert after the branch-3.2 cut? I believe we should not land this kind of PR in `master` branch. Instead, we can have this in `branch-3.2` if needed for those correctness issue. WDYT, @mridulm and @otterc ?

**Comment by mridulm** [comment]
That makes sense, let us close this PR @otterc and create it against branch-3.2 when it has been cut and if we determine that the two correctness fixes cant make it to 3.2

**Comment by HyukjinKwon** [comment]
cc @Ngone51 FYI

**Comment by HyukjinKwon** [comment]
Yeah, +1 for landing this to branch-3.2 only

**Comment by HyukjinKwon** [comment]
only few days left for the branchcut :-).

**Comment by otterc** [comment]
Makes sense @dongjoon-hyun @mridulm @HyukjinKwon. Will close this PR and create it against 3.2 once it is cut.

**Comment by Ngone51** [comment]
I'm thinking that if we dynamically disable push-based shuffle when indeterminated stage retries (e.g., via job properties), users are still safe to try it? 

(Note that push-based shuffle is already disabled in the case of multiple yarn attempts in master branch)

**Comment by mridulm** [comment]
@Ngone51 The problem is that fixing the two correctness pending issues requires protocol changes - which means serde issues if 3.3 (where this is fixed) tries to work with 3.2 (if it is released without the fixes) - client/ESS combinations (3.3 client with 3.2 ESS and 3.2 client with 3.3 ESS in future).

**Comment by Ngone51** [comment]
I see. Make sense to me.

**Pull request closed without merging**
